Upstream-Status: Backport

nm: Fix size passed to snprintf for invalid sh_name case.
native build failed as following on Fedora18:
nm.c: In function 'show_symbols_sysv':
nm.c:756:27: error: argument to 'sizeof' in 'snprintf' call is the same expression as the destination; did you mean to provide an explicit length? [-Werror=sizeof-pointer-memaccess]
    snprintf (name, sizeof name, "[invalid sh_name %#" PRIx32 "]",
                           ^

The original commit is http://git.fedorahosted.org/cgit/elfutils.git/commit/src/nm.c?id=57bd66cabf6e6b9ecf622cdbf350804897a8df58

Signed-off-by: Zhenhua Luo <zhenhua.luo@freescale.com>

--- elfutils-0.148/src/nm.c.org	2013-03-11 22:36:11.000000000 -0500
+++ elfutils-0.148/src/nm.c	2013-03-11 22:46:09.000000000 -0500
@@ -752,8 +752,9 @@
 			       gelf_getshdr (scn, &shdr_mem)->sh_name);
       if (unlikely (name == NULL))
 	{
-	  name = alloca (sizeof "[invalid sh_name 0x12345678]");
-	  snprintf (name, sizeof name, "[invalid sh_name %#" PRIx32 "]",
+	  const size_t bufsz = sizeof "[invalid sh_name 0x12345678]";
+	  name = alloca (bufsz);
+	  snprintf (name, bufsz, "[invalid sh_name %#" PRIx32 "]",
 		    gelf_getshdr (scn, &shdr_mem)->sh_name);
 	}
       scnnames[elf_ndxscn (scn)] = name;
